homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,132 @@
 from typing import Any
 
 from aiohttp import ClientConnectionError
-from intellifire4py import IntellifireAsync
+from intellifire4py import AsyncUDPFireplaceFinder, IntellifireAsync
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_HOST
-from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResult
 
-from .const import DOMAIN
+from .const import DOMAIN, LOGGER
 
 STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
 
 
-async def validate_input(hass: HomeAssistant, host: str) -> str:
+async def validate_host_input(host: str) -> str:
     """Validate the user input allows us to connect.
 
     Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
     """
     api = IntellifireAsync(host)
     await api.poll()
-
+    ret = api.data.serial
+    LOGGER.debug("Found a fireplace: %s", ret)
     # Return the serial number which will be used to calculate a unique ID for the device/sensors
-    return api.data.serial
+    return ret
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for IntelliFire."""
 
     VERSION = 1
 
-    async def async_step_user(
+    def __init__(self):
+        """Initialize the Config Flow Handler."""
+        self._config_context = {}
+        self._discovered_hosts: list[str] = []
+
+    async def _find_fireplaces(self):
+        """Perform UDP discovery."""
+        fireplace_finder = AsyncUDPFireplaceFinder()
+        self._discovered_hosts = await fireplace_finder.search_fireplace(timeout=1)
+
+    async def async_step_local_config(self, user_input=None):
+        """Handle local ip configuration."""
+
+        errors = {}
+        placeholder = {}
+        local_schema = vol.Schema({vol.Required(CONF_HOST): str})
+
+        if user_input is None:
+
+            current_hosts = [
+                entry.data[CONF_HOST]
+                for entry in self._async_current_entries(include_ignore=False)
+            ]
+
+            if self._discovered_hosts != []:
+                # Filter out already configured hosts
+                self._discovered_hosts = [
+                    ip for ip in self._discovered_hosts if ip not in current_hosts
+                ]
```

Comment:
It is confusing to mutate `self._discovered_hosts` and remove discoveries because they have already been setup.  It would be cleaner to create a new variable with unconfigured_discovered_hosts

Reply:
I guess I can move the filtering as well to the discovery stage ... so this is a good suggestion!